### ISSUE TITLE
Complete EntityDataTypes(Utilize WrappedBlockState and Particle)

### ISSUE THREAD

**Issue by retrooper** [retrooper/packetevents#180]

**Comment by retrooper** [comment]
@MWHunter Is it safe for us to use WrappedBlockState in EntityDataTypes already? You tell me.

Later on we'll work on Particles, as thats the only thing missing.

**Comment by retrooper** [comment]
Made particle abstraction, isn't entirely done yet. But works on 1.18.1, that is for sure.

**Comment by MWHunter** [comment]
I disagree with automatically parsing WrappedBlockState due to the additional memory this would add when loading this class.
